cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -113,8 +113,19 @@ public void configure(Map<String, ?> configs) {
 
     //Add BootstrapServers if not set
     if (!producerProps.containsKey(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG)) {
-      Object port = configs.get("port");
-      String bootstrapServers = "localhost:" + (port == null ? "9092" : String.valueOf(port));
+      String listeners = configs.get("listeners") == null ? null : String.valueOf(configs.get("listeners"));
+      String bootstrapServers = null;
+      if (listeners != null && listeners.length() != 0) {
+        String firstListener = listeners.split("\\s*,\\s*")[0];
+        String[] allListeners = firstListener.split(":");
+        if (allListeners[1].length() == 2) {
+          bootstrapServers = "0.0.0.0:" + allListeners[2];
+        } else {
+          bootstrapServers = allListeners[1].substring(2, allListeners[1].length()) + ":" + allListeners[2];
+        }
+      } else {
+        bootstrapServers = "localhost:9092";
+      }
```

Comment:
Thanks for the update!
https://github.com/linkedin/cruise-control/pull/1246#discussion_r448045320 from the first round of review seems not addressed. This patch drops parsing the value of `port`. However, existing users might depend on value retrieval via `port` config.

1. Can we make sure that the behavior is the same as before if `port` config exists?
2. Can we move this parsing logic to a small function with this priority of retrieval documented?

-- please let me know if you have any questions.

Reply:
Thanks for your answer. l know,let me modify.

Reply:
If `KafkaConfig.ListenersProp()` is configured, `port ` is not null and the value of port will not be used，maybe

1. `if configs.get(KafkaConfig.ListenersProp()) ` is not null, parse host and port in KafkaConfig.ListenersProp(),use this as port
2. otherwise `if configs.get(KafkaConfig.PortProp())`  is not null , use that as port
3. otherwise use the default hostname and port -- i.e. localhost and 9092.